fabric-registry-sync-v0/src/main/java/net/fabricmc/fabric/impl/registry/sync/RegistrySyncManager.java

```diff
@@ -304,4 +317,14 @@ public static void unmap() throws RemapException {
 	public static void bootstrapRegistries() {
 		postBootstrap = true;
 	}
+
+	public static void addServerOnlyRegistryId(Registry<?> registry, Identifier identifier) {
+		Set<Identifier> identifiers = SERVER_ONLY_REGISTRY_IDS.computeIfAbsent(registry, (r) -> new HashSet<>());
+
+		if (identifiers.contains(identifier)) {
+			throw new IllegalStateException(identifier + " has already been been marked as server only in " + registry);
```

Comment:
No need to be this harsh. Imo a log warning is enough as if an id is added twice there won't be any behavioral changes